api/v1beta1/kafkacluster_types.go

```diff
@@ -436,6 +444,12 @@ type CommonListenerSpec struct {
 	// +kubebuilder:validation:Enum=ssl;plaintext;sasl_ssl;sasl_plaintext
 	Type SecurityProtocol `json:"type"`
 	// +kubebuilder:validation:Pattern=^[a-z0-9\-]+
+	// ServerSSLCertSecretName is where custom server SSL certificate can be provided.
+	// Secret has to contain the keystore and truststore in jks and the password for them in base64 encoded format format.
+	// Data fields must be: keystore.jks, truststore.jks, password
+	// SecurityProtocol has to be set to ssl.
+	ServerSSLCertSecretName string `json:"serverSSLCertSecretName,omitempty"`
```

Comment:
Since this is a reference to a Kubernetes resource in the same namespace the type of this field could be `corev1.LocalObjectReference` to be in line with how Kubernetes resources references other resources in the same namespace.